overlord/devicestate/handlers_install.go

```diff
@@ -1278,7 +1295,19 @@ func (m *DeviceManager) doInstallSetupStorageEncryption(t *state.Task, _ *tomb.T
 		return fmt.Errorf("reading gadget information: %v", err)
 	}
 
-	encryptInfo, err := m.encryptionSupportInfo(systemAndSeeds.Model, secboot.TPMProvisionFull, systemAndSeeds.InfosByType[snap.TypeKernel], gadgetInfo)
+	var systemSnapdVersions *installLogic.SystemSnapdVersions
+	// Find snapd versions for snapd and kernel snaps in the seed for
+	// the passphrase/PINs auth checks.
+	// FDE is only supported in UC20+ (i.e. Model grade is set).
+	if systemAndSeeds.Model.Grade() != asserts.ModelGradeUnset {
```

Comment:
I'm actually not sure we need the check at this point because we wouldn't have made this far if the system is not UC20+

Reply:
though it would be interesting to double check what kind of errors /v2/systems/foo gives on UC18/classic system

Reply:
Thank you! I didn't know that UC18 is not even installable through the systems API, and indeed `snap prepare-image` for a UC18 model doesn't create the `system-seed` directory.

Reply:
Removed restriction, Now the snapd versions structure is populated `loadAndMountSystemLabelSnaps` and embedded inside `systemAndEssentialSnaps`.